### PR TITLE
Setting and getting the pr status, with firebase and server and creating a pr on github

### LINKED ISSUE #81: Sync PR status with Firebase

### PR DESCRIPTION
**Pull request by eanker**
In order to make this work use preview-code/backend#2

Fix #81

### CODE REVIEW

**Comment by eanker** [comment]
Should I collapse the two iron-ajax calls in create-pull-request, and see if I can call the setting of the status inside the creating of the pull request?

File: src/projects/create-pull-request/create-pull-request.html [hunk 1]
```diff
@@ -167,6 +199,28 @@
       _showPullRequestInformation: function() {
         this.$.diffView.show();
         this.$.description.show();
+      },
+      _createPullRequest: function() {
+        console.log(this.description);
```
Comment (TimvdLippe) [inline src/projects/create-pull-request/create-pull-request.html]: Left-over debug statement.

File: src/projects/create-pull-request/create-pull-request.html [hunk 2]
```diff
@@ -167,6 +199,28 @@
...
+        this.$.createPull.generateRequest();
+      },
+      _updateBackendPathCreate: function(title, headBranch, baseBranch) {
+        return 'https://localhost:8443/project/createPR' + '/' + headBranch + '/' + baseBranch;
+      },
+      _updateBackendPathStatus: function(project, number) {
+        return 'https://localhost:8443/pr/' + this.project.owner + '/' + this.project.name + '/' + number + '/status';
+      },
+      _setStatus: function() {
+        if(this.number !== 0) {
```
Comment (TimvdLippe) [inline src/projects/create-pull-request/create-pull-request.html]: Space between  `if` and `(`

**Comment by TimvdLippe** [comment]
The backend should set the status instead. So when the backend finished contacting GitHub and it gets an ID back, it should post to Firebase with the project and the number. Then the frontend will automatically be updated. The second iron-ajax is only required when you are manually setting a PR status in an already created PR.

File: src/projects/create-pull-request/create-pull-request.html [hunk 3]
```diff
@@ -91,6 +91,23 @@
...
+        last-response="{{number}}"
+        on-response="_setStatus"
+        ></iron-ajax>
+
+      <iron-ajax
+        id="setPrStatusafterCreate"
+        method="POST"
+        url="[[backendPathStatus]]"
+        headers='{"Content-Type": "application/json"}'
+        handle-as="json"></iron-ajax>
```
Comment (txsmith) [inline src/projects/create-pull-request/create-pull-request.html]: This element seems to be duplicate in `pr-status.html`. Can we maybe reuse that element for setting the status?

**Comment by txsmith** [comment]
Two general remarks/questions:
My first question is why the PR status is set in a request separate from creating the PR. Is there a reason why this cannot be done in one call?
My second question: if we really want to set the status in a separate request. The duplicate code in `create-pull-request` and `pr-status` should be eliminated. If the status is set separately, there should be just one element responsible for this.

**Comment by TimvdLippe** [comment]
@txsmith Yes that was what I was trying to describe in my previous comment too. @eanker and I talked about it and it will be fixed.

**Comment by txsmith** [comment]
I totally missed your comment, sorry!

**Comment by TimvdLippe** [comment]
LGTM

**Pull request merged**